isthmus/src/main/java/io/substrait/isthmus/SqlExpressionToSubstrait.java

```diff
@@ -140,10 +142,10 @@ private Result registerCreateTablesForExtendedExpression(List<String> tables)
     CalciteSchema rootSchema = CalciteSchema.createRootSchema(false);
     CalciteCatalogReader catalogReader =
         new CalciteCatalogReader(rootSchema, List.of(), factory, config);
-    SqlValidator validator = Validator.create(factory, catalogReader, SqlValidator.Config.DEFAULT);
     if (tables != null) {
       for (String tableDef : tables) {
-        List<SubstraitTable> tList = parseCreateTable(factory, validator, tableDef);
+        List<SubstraitTable> tList =
+            SubstraitCreateStatementParser.processCreateStatements(tableDef);
```

Comment:
this method might (in a future PR) also be a candidate for refactoring. I'm wondering if we really need the variant of `processCreateStatements()` returning a list of Substrait tables to be public or whether the logic of the `registerCreateTablesForExtendedExpression` method could be rewritten to work with the `CatalogReader` instead of the list of tables.